### DISCUSSION TITLE
logfile including herdsman debug logs

### DISCUSSION THREAD

**Discussion by ascheucher** [Koenkk/zigbee2mqtt#11896] [General]
Maybe I am just blind, but I couldn't find a file which includes the herdsman debug logs. I only found them in the Home Assistant ZigBee2MQTT Supervisor GUI's log view.

The logfiles e.g. /config/zigbee2mqtt/log/2022-03-18.09-58-11/log.txt don't contain them.

As the Suervisor GUI's log view is rather limited in line counts, I really would like to have them in a file somewhere when possible.

Thanks for your input! 

Edit: OK. I am blind, found it: https://www.zigbee2mqtt.io/guide/usage/debug.html#zigbee-herdsman-debug-logging

**Comment by cloudbr34k84** [comment]
> Maybe I am just blind, but I couldn't find a file which includes the herdsman debug logs. I only found them in the Home Assistant ZigBee2MQTT Supervisor GUI's log view.
> 
> The logfiles e.g. /config/zigbee2mqtt/log/2022-03-18.09-58-11/log.txt don't contain them.
> 
> As the Suervisor GUI's log view is rather limited in line counts, I really would like to have them in a file somewhere when possible.
> 
> Thanks for your input!
> 
> Edit: OK. I am blind, found it: https://www.zigbee2mqtt.io/guide/usage/debug.html#zigbee-herdsman-debug-logging

how do you get them as the instructions don't make sense to me? Im running Z2m ADD On and have enabled the  herdsman debug logs, which I can see on the Tab on the Z2M add on page, but I cant actually see how to get the log file
![image](https://github.com/Koenkk/zigbee2mqtt/assets/58960644/fe78f9f0-46ce-4ce6-ae48-537ab99dd918)

**Reply by nick4275** [reply]
Did you ever sort this ? - I've got an issue at the moment where I want to view the logs (as you did) but I want to extend the length of the log to a few days. I can get the log exported via a few console commands but can't get the length of the file to span a few days. Wondered if you came across the answer to this in your research
